src/features/voting/ProposalList.tsx

```diff
@@ -5,60 +5,85 @@ import ProposalFormButton from './ProposalFormButton.tsx';
 import ProposalCard, { type ProposalCardProps } from './ProposalCard.tsx';
 import { Button } from '../ui/button.tsx';
 import { sdk, type Paginated } from '../../sdk.ts';
+import { useSession } from '../auth/hooks.ts';
+import { LoadingSpinner } from '../ui/LoadingSpinner.tsx';
 
 // This component auto-loads proposals on scroll, so we hard-code a static limit
 const limit = 10;
 
 export function ProposalList() {
-  const [loading, setLoading] = useState(false);
+  const session = useSession();
+  const [loading, setLoading] = useState(true);
   const [cursor, setCursor] = useState<Paginated['cursor']>();
   const [proposals, setProposals] = useState<ProposalCardProps[]>([]);
 
-  const loadProposals = useCallback((pagination: Paginated) => {
-    setLoading(true);
-    sdk
-      .listProposals({ queries: { pagination } })
-      .then((result) => {
-        setCursor(result.cursor);
-        setProposals((previous) => [...previous, ...result.proposals]);
-      })
-      .catch(toast.error)
-      .finally(() => setLoading(false));
-  }, []);
+  const loadProposals = useCallback(
+    async (pagination: Paginated) => {
+      const token = await session?.getToken();
+
+      setLoading(true);
+
+      sdk
+        .listProposals({
+          queries: { pagination },
+          headers: token ? { Authorization: `Bearer ${token}` } : {},
+        })
+        .then((result) => {
+          setCursor(result.cursor);
+          setProposals((previous) => [...previous, ...result.proposals]);
+        })
+        .catch(toast.error)
+        .finally(() => setLoading(false));
+    },
+    [session],
+  );
```

Comment:
QUESTION: Are we putting proposals behind an auth wall? I was under the impression that proposals are public, but not vote results. If the idea is to check the token so we can display the correct vote results for open proposals, perhaps we should instead display a message prompting the user to log in or something, instead of refusing to load proposals at all?

NOTE: Note that our button components actually have a lot of functionality. There is a `ProtectedTrigger` component that "just works" if logged in, but if not displays the loggin modal instead. It also has a loading animation when the `loading` prop is true (we have a loading spinner library for standardized spinners, see the button for implementation). Do we need another loading animation, or can we just disable that button when there are no more to load?

Reply:
Hmm... I see... Apparently our loading spinner component quit working?

Reply:
After investigating, looks like the loading spinner component chokes when it tries to render on the server, which it tries to do now that we have the initial state of `loading` set to `true`. Since the default state is true, it loads the spinner on first render, which happens on the server... instead of waiting until the client code runs which sets it to 'loading=true').

Setting it to `false` causes a flash of content, though. Not sure which is better here really... I think the ideal solution probably has something to do with using suspense with fallback correctly. 

I guess the main thing is to be sure we've got the semantics we're going for. I'm not hung up on the spinner as long as we're intentional about who can see what when logged in.